Build failure in optionsx expansion with _Bool with clang 17 on Mac

On my Mac with clang 17 I see this error building options.c:

```
$ ninja dynamorio
[13/96] Building C object core/CMakeFiles/dynamorio.dir/options.c.o
FAILED: [code=1] core/CMakeFiles/dynamorio.dir/options.c.o
/usr/bin/cc -DDYNAMORIO_INTERNAL -D_LARGEFILE64_SOURCE -Ddynamorio_EXPORTS -Idr/git/src/core/drlibc -Idr/git/src/core/ir/aarch64 -Idr/git/src/core/arch/aarch64 -Idr/git/src/core/unix -Idr/git/src/core/ir -Idr/git/src/core/arch -Idr/git/src/core/lib -Idr/git/build_m1_dbg -fno-strict-aliasing -funsigned-char -fno-stack-protector -mmacosx-version-min=11.7 -fvisibility=hidden  -std=gnu99 -g3 -fno-omit-frame-pointer  -Wall -Werror -Wwrite-strings -Wvla -Wno-unused-but-set-variable -Wno-deprecated-non-prototype -Wno-asm-operand-widths -arch arm64 -fPIC -MD -MT core/CMakeFiles/dynamorio.dir/options.c.o -MF core/CMakeFiles/dynamorio.dir/options.c.o.d -o core/CMakeFiles/dynamorio.dir/options.c.o -c dr/git/src/core/options.c
In file included from dr/git/src/core/options.c:211:
dr/git/src/core/optionsx.h:217:1: error: use of undeclared identifier 'OPTION_TYPE__Bool'; did you mean 'OPTION_TYPE_bool'?
  217 | DYNAMIC_OPTION_DEFAULT(bool, dynamic_options, true, "dynamically update options")
      | ^
dr/git/src/core/optionsx.h:100:5: note: expanded from macro 'DYNAMIC_OPTION_DEFAULT'
  100 |     OPTION_COMMAND(type, name, value, #name, {}, description, DYNAMIC, OP_PCACHE_NOP)
      |     ^
dr/git/src/core/options.c:206:7: note: expanded from macro 'OPTION_COMMAND'
  206 |       OPTION_TYPE_##type,                                                         \
      |       ^
<scratch space>:112:1: note: expanded from here
  112 | OPTION_TYPE__Bool
      | ^
dr/git/src/core/options.c:111:5: note: 'OPTION_TYPE_bool' declared here
  111 |     OPTION_TYPE_bool,
      |     ^
In file included from dr/git/src/core/options.c:211:
dr/git/src/core/optionsx.h:229:1: error: use of undeclared identifier 'OPTION_TYPE__Bool'; did you mean 'OPTION_TYPE_bool'?
  229 | PC_OPTION_INTERNAL(bool, nolink, "disable linking")
...
```

The error repeats a bunch of times.

It's coming from some `#define` of `bool` to be `_Bool` in some system header I think.
Undefining at top level didn't help; I had to undefine inside optionsx.h.

--- a/core/optionsx.h
+++ b/core/optionsx.h
@@ -1,5 +1,5 @@
 /* *******************************************************************************
- * Copyright (c) 2010-2025 Google, Inc.  All rights reserved.
+ * Copyright (c) 2010-2026 Google, Inc.  All rights reserved.
  * Copyright (c) 2011 Massachusetts Institute of Technology  All rights reserved.
  * Copyright (c) 2003-2010 VMware, Inc.  All rights reserved.
  * *******************************************************************************/
@@ -66,6 +66,11 @@
 /* The liststring_t type is assumed to contain ;-separated values that are
  * appended to if multiple option instances are specified
  */
+
+#ifdef bool
+/* Avoid problems with bool macros on Mac. */
+#    undef bool
+#endif
 
 /* Shortcuts for the common cases */
 #define OPTION_DEFAULT(type, name, value, description) \

